src/librustdoc/clean/types.rs

```diff
@@ -870,8 +871,10 @@ pub(crate) enum ItemKind {
     /// A required associated constant in a trait declaration.
     RequiredAssocConstItem(Generics, Box<Type>),
     ConstantItem(Box<Constant>),
-    /// An associated constant in a trait impl or a provided one in a trait declaration.
-    AssocConstItem(Box<Constant>),
+    /// An associated constant in a trait declaration with provided default value.
+    DefaultAssocConstItem(Box<Constant>),
```

Comment:
`DefaultAssocConstItem` sounds like `default const ITEM: Ty = EXPR` (`{,min_}specialization`) to me. Maybe `ProvidedAssocConstItem` or `DefaultedAssocConstItem` instead?

Reply:
None of the terminology is fantastic, but I'm okay with either of your suggestions; they are both improvements on what is in the PR. "Provided" is probably the least confusable. Initially it sounds like it could possibly refer to "the constant value that a **trait impl** provided as the value for a trait's associated constant" (provided by impl vs provided by trait) but it doesn't seem possible to remain confused about this for long given the existence of `ImplAssocConstItem` which clearly refers to something in an impl.

I was also thinking along the lines of `OptionalAssocConstItem` to contrast with `RequiredAssocConstItem` or even synonyms like "discretionary" but none of that is established terminology in Rust in this context.